### PR TITLE
Add new standard for Global Quarantine Notification Interval 

### PR DESCRIPTION
**Pull request by kris6673**
API part for https://github.com/KelvinTegelaar/CIPP/pull/2397

### CODE REVIEW

**Comment by JohnDuprey** [comment]
close for hotfix

File: Modules/CIPPCore/Public/Standards/Invoke-CIPPStandardGlobalQuarantineNotifications.ps1 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    # but i just cant understand why the API returns the values it does and not a timespan like the equivalent powershell command does
+    # If you know why, please let me know -Bobby
+    $CurrentState.EndUserSpamNotificationFrequency = switch ($CurrentState.EndUserSpamNotificationFrequency) {
+        'PT4H' { New-TimeSpan -Hours 4 }
+        'P1D' { New-TimeSpan -Days 1 }
+        'P7D' { New-TimeSpan -Days 7 }
+        Default { $null }
+    }
+
+    if ($Settings.remediate) {
```
Comment (OfficialEsco) [inline Modules/CIPPCore/Public/Standards/Invoke-CIPPStandardGlobalQuarantineNotifications.ps1:29]:
```suggestion
    if ($Settings.remediate -eq $true) {
```

File: Modules/CIPPCore/Public/Standards/Invoke-CIPPStandardGlobalQuarantineNotifications.ps1 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                New-ExoRequest -tenantid $Tenant -cmdlet 'Set-QuarantinePolicy' -cmdParams @{Identity = $CurrentState.Identity; EndUserSpamNotificationFrequency = [string]$WantedState } -useSystemmailbox $true
+                Write-LogMessage -API 'Standards' -tenant $tenant -message "Set Global Quarantine Notifications to $WantedState" -sev Info
+            } catch {
+                $ErrorMessage = Get-NormalizedError -Message $_.Exception.Message
+                Write-LogMessage -API 'Standards' -tenant $tenant -message "Failed to set Global Quarantine Notifications to $WantedState. Error: $ErrorMessage" -sev Error
+            }
+        }
+    }
+
+    if ($Settings.alert) {
```
Comment (OfficialEsco) [inline Modules/CIPPCore/Public/Standards/Invoke-CIPPStandardGlobalQuarantineNotifications.ps1:45]:
```suggestion
    if ($Settings.alert -eq $true) {
```

File: Modules/CIPPCore/Public/Standards/Invoke-CIPPStandardGlobalQuarantineNotifications.ps1 [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        
+        if ($CurrentState.EndUserSpamNotificationFrequency -eq $WantedState) {
+            Write-LogMessage -API 'Standards' -tenant $tenant -message "Global Quarantine Notifications are set to the desired value of $WantedState" -sev Info
+        } else {
+            Write-LogMessage -API 'Standards' -tenant $tenant -message "Global Quarantine Notifications are not set to the desired value of $WantedState" -sev Alert
+        }
+    }
+
+    if ($Settings.report) {
```
Comment (OfficialEsco) [inline Modules/CIPPCore/Public/Standards/Invoke-CIPPStandardGlobalQuarantineNotifications.ps1:54]:
```suggestion
    if ($Settings.report -eq $true) {
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]
They have added $true check for these since https://github.com/KelvinTegelaar/CIPP-API/pull/787 (which makes sense if it comes back as $false)

**Comment by kris6673** [comment]
@OfficialEsco Must have missed that somehow, thanks!

**Comment by OfficialEsco** [comment]
2 days ago tho, just noticed it because of a merge conflict :smile:

**Comment by KelvinTegelaar** [comment]
Correct, brand spanking new :)

**Comment by KelvinTegelaar** [comment]
Thanks guys! Great work.

**Comment by kris6673** [comment]
> Thanks guys! Great work.

Just ignore line 19 to 27 🤡

**Comment by KelvinTegelaar** [comment]
hahahaha, so to explain that one, as MS doesn't: PT time is what the Defender and Graph APIs use to define a timespan without needing to do math yourself. They use this in a couple of places where it's just easier to have them figure it out.

**Comment by kris6673** [comment]
> hahahaha, so to explain that one, as MS doesn't: PT time is what the Defender and Graph APIs use to define a timespan without needing to do math yourself. They use this in a couple of places where it's just easier to have them figure it out.

Oh that makes sense to save on computing wherever you can
If you can think of a better way to parse it, do let me know ^^

**Pull request merged**